### PR TITLE
Issue #55 Improve report part

### PR DESCRIPTION
**Pull request by eunyyyoung**
#55 
 To send the error code will be resolved by another issue (#60) .

### CODE REVIEW

File: Backend/router/v1/report.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = () => {
   });
 
   router.all('/', (req, res) => {
-    res.status(404).send("This is not normal route");
+    res.status(404).json({msg: "This is not normal route"});
```
Comment (mtnviewlab) [inline Backend/router/v1/report.js]: This is a very kind msg to a  hacker. Why don't you send a status code instead and let the client handles it? Do not specify what is wrong in this way.

File: Backend/router/v1/report.ctrl.js [hunk 2]
```diff
@@ -1,120 +1,122 @@
...
   }
+  models.report.create({
+    what: req.body.data[0].what,
+    location: req.body.data[0].location,
+    time: req.body.data[0].time,
+    who: req.body.data[0].who,
+    details: req.body.data[0].details,
+    userID: req.user[0].dataValues.id
+  }).then((result) => {
+    res.status(201);
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: I think you can combine this line with the below one.

File: Backend/router/v1/report.ctrl.js [hunk 3]
```diff
@@ -1,120 +1,122 @@
...
+    res.json({id: result.dataValues.id});
+  }).catch((err) => {
+    res.json({msg: "Error: failed to create report"});
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: Is there a status code for this error case?

File: Backend/router/v1/report.ctrl.js [hunk 4]
```diff
@@ -1,120 +1,122 @@
...
-      res.json(result);
-    }).catch((err) => {});
-  } else {
-    res.status(401)
-    res.json({
-      msg: "get report failed"
-    })
+  if (!req.isAuthenticated()) {
+    res.status(403).json({msg: "You are not logged in"});
+    return ;
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: remove extra space

File: Backend/router/v1/report.ctrl.js [hunk 5]
```diff
@@ -1,120 +1,122 @@
...
   }
+  models.report.findAll({
+    where: {
+      userID: req.user[0].dataValues.id
+    }
+  }).then((result) => {
+    res.status(200);
+    res.json(result);
+  }).catch((err) => {
+    res.json({msg: "Error: failed to load all reports"});
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: Same here, is there a status code for this error case?

Comment (shberry) [reply Backend/router/v1/report.ctrl.js]: Same below.

File: Backend/router/v1/report.ctrl.js [hunk 6]
```diff
@@ -1,120 +1,122 @@
...
+      res.status(204).end();
+    }
+  }).catch((err) => {
+    res.json({msg: "Error: failed to delete this report"});
+  });
 };
 
 const editReport = (req, res) => {
+  if (!req.isAuthenticated()) {
+    res.status(403).json({msg: "You are not logged in"});
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: Can you update the error message so it's more consistent with "Error: failed to delete this report"?

File: Backend/router/v1/report.ctrl.js [hunk 7]
```diff
@@ -1,120 +1,122 @@
...
+    },{
+      where: {id: reportId}
+  }).then((result) => {
+    if (result == 0) {
+      res.json({msg: "The reportId does not exist"});
+    } else {
+      res.status(200).end();
+    }
+  }).catch((err) => {
+    res.json({msg: "Error: failed to edit this repart"});
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: Same here, should we have a status code for this error case?

File: Backend/router/v1/report.ctrl.js (hunk 3 above)
Comment (eunyyyoung) [reply Backend/router/v1/report.ctrl.js]: There is no http error code. The application error code will be handled by another issue soon. ( #60 )

File: Backend/router/v1/report.js (hunk 1 above)
Comment (eunyyyoung) [reply Backend/router/v1/report.js]: The application error code will be handled by another issue soon. ( #60 )

File: Backend/router/v1/report.ctrl.js (hunk 5 above)
Comment (eunyyyoung) [reply Backend/router/v1/report.ctrl.js]: There is no http error code. The application error code will be handled by another issue soon. ( #60 )

File: Backend/router/v1/report.ctrl.js (hunk 6 above)
Comment (eunyyyoung) [reply Backend/router/v1/report.ctrl.js]: I don't know what error is caught and I think this is enough. @shberry

File: Backend/router/v1/report.ctrl.js (hunk 7 above)
Comment (eunyyyoung) [reply Backend/router/v1/report.ctrl.js]: humm... I think we don't need it. It will be just sending an error code to client.

**Comment by eunyyyoung** [comment]
I modified it to the same branch without creating a new branch for Issue #60.

**Pull request merged**